File: Java Barcode Scanner ME/app/src/main/java/de/dimedis/mobileentry/util/UtilsService.java
package de.dimedis.mobileentry.util;

import android.content.Context;
import android.content.Intent;

import androidx.annotation.NonNull;
import androidx.work.Worker;
import androidx.work.WorkerParameters;

import de.dimedis.mobileentry.ui.InitActivity;

public class UtilsService extends Worker {
    public UtilsService(@NonNull Context context, @NonNull WorkerParameters params) {
        super(context, params);
    }

    @NonNull
    @Override
    public Result doWork() {
        performCleanupTasks();
        return Result.success();
    }

    private void performCleanupTasks() {
        FileUtils.clearApplicationData(getApplicationContext());
        Intent intentNew = new Intent(getApplicationContext(), InitActivity.class);
        intentNew.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
        getApplicationContext().startActivity(intentNew);
    }
}

File: Kotlin Migration/NTMobileEntry/app/src/main/java/de/dimedis/mobileentry/util/UtilsService.kt
package de.dimedis.mobileentry.util

import android.content.Context
import android.content.Intent
import android.os.Process
import androidx.work.Worker
import androidx.work.WorkerParameters
import de.dimedis.mobileentry.ui.InitActivity

class UtilsService(context: Context, params: WorkerParameters) : Worker(context, params) {
    override fun doWork(): Result {
        val pid = Process.myPid()
        Process.killProcess(pid)
        FileUtils.clearApplicationData(applicationContext)
        val intentNew = Intent(applicationContext, InitActivity::class.java)
        intentNew.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK)
        applicationContext.startActivity(intentNew)
        return Result.success()
    }
}